bin/main.py

```diff
@@ -52,6 +53,15 @@ def run():
         else:
             Thread(target=process, args=(line, None)).start()
 
+def run_ipc():
+    receive_signal = PluginLoader('ipc_plugin.ipc_mod').receive_signal
+    try:
+        while True:
+            signal, value = receive_signal()
+            Thread(target=process, args=(signal, value)).start()
```

Comment:
You're not dealing with sending here (the sending bit is currently the function that logs a message - it should probably be split from logging and renamed to send).